Src/PerformanceCollector/Shared/Implementation/QuickPulse/Process/PerfLib/PerfLib.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.PerfLib
+{
+    using System;
+    using System.Globalization;
+
+    internal class PerfLib : IQuickPulsePerfLib
+    {
+        private static PerfLib library = null;
+
+        private PerformanceMonitor performanceMonitor;
+
+        private PerfLib()
+        {
+        }
+
+        public static PerfLib GetPerfLib()
+        {
+            library = library ?? new PerfLib();
```

Comment:
Not sure what you mean here. If instead of lazy initialization we do a regular one - we'd still have a static instance to store the object in, right? Regardless, for our usage it doesn't really matter, we don't call this 100 times in a loop or anything, so any approach which is functionally correct should be fine.